### PR TITLE
Move trace category settings to timeline events tab bar

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This addresses some feedback from @zanderso regarding the trace category settings not being discoverable. Moving these settings closer to the feature they pertain to should help with this.

Screenshot 2023-03-20 at 7 10 57 PM
Screenshot 2023-03-20 at 7 07 09 PM
Screenshot 2023-03-20 at 7 04 10 PM

The Performance Settings dialog now looks like this:
Screenshot 2023-03-20 at 7 12 31 PM

This PR does some yak shaving to refactor code and group Timeline Events widgets together in timeline_events_view.dart, and also fixes a couple render flex overflow errors.

RELEASE_NOTE_EXCEPTION=[too small to mention]

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    return DevToolsDialog(
+      title: const DialogTitleText('Trace Categories'),
+      includeDivider: false,
+      content: Container(
+        width: defaultDialogWidth,
+        child: Column(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart:174]: from here down, this was moved from performance_settings.dart

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    unawaited(
+      showDialog(
+        context: context,
+        builder: (context) => TraceCategoriesDialog(controller),
+      ),
+    );
+  }
+}
+
+class RefreshTimelineEventsButton extends StatelessWidget {
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart:133]: moved from tabbed_performance_view.dart

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+}
+
+class TimelineEventsTabControls extends StatelessWidget {
+  const TimelineEventsTabControls({super.key, required this.controller});
+
+  final TimelineEventsController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart:65]: moved from tabbed_performance_view.dart

File: packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+import 'timeline_events_controller.dart';
+
+class TimelineEventsTabView extends StatelessWidget {
+  const TimelineEventsTabView({super.key, required this.controller});
+
+  final TimelineEventsController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<bool>(
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/performance/panes/timeline_events/timeline_events_view.dart:29]: moved from tabbed_performance_view.dart

**Review by bkonyi — APPROVED** [review]

**Pull request merged**